terminado/tests/basic_test.py

```diff
@@ -5,42 +5,223 @@
+                                        ioloop=self.io_loop)
+        single_tm = SingleTermManager(shell_command=['bash'], ioloop=self.io_loop)
+        unique_tm = UniqueTermManager(shell_command=['bash'], max_terminals=MAX_TERMS,
+                                        ioloop=self.io_loop)
+        
+        class NewTerminalHandler(tornado.web.RequestHandler):
+            """Create a new named terminal, return redirect"""
+            def get(self):
+                name, terminal = named_tm.new_named_terminal()
+                self.redirect("/named/" + name, permanent=False)
+
+        return tornado.web.Application([
+                    (r"/new",         NewTerminalHandler),
+                    (r"/named/(\w+)", TermSocket, {'term_manager': named_tm}),
+                    (r"/single",      TermSocket, {'term_manager': single_tm}),
+                    (r"/unique",      TermSocket, {'term_manager': unique_tm})
+                ], debug=True)
+
+    test_urls = ('/named/term1', '/unique', '/single')
+
+class CommonTests(TermTestCase):
     @tornado.testing.gen_test
     def test_basic(self):
-        ws = yield self.ws_connect("/websocket/term1")
-        response = yield ws.read_message()              # Setup message
-        j = json.loads(response)
-        self.assertEqual(j, ['setup', {}])
-
-        response = yield ws.read_message()              # Command prompt
-        j = json.loads(response)
-        self.assertEqual(j[0], 'stdout')
-        ws.close()
+        for url in self.test_urls:
+            tm = yield self.get_term_client(url)
+            response = yield tm.read_msg()                    
+            self.assertEqual(response, ['setup', {}])
+
+            # Check for initial shell prompt
+            response = yield tm.read_msg()
+            self.assertEqual(response[0], 'stdout')
+            self.assertGreater(len(response[1]), 0)
+            tm.close()
+
+    @tornado.testing.gen_test
+    def test_basic_command(self):
+        for url in self.test_urls:
+            tm = yield self.get_term_client('/named/foo')
+            yield tm.read_all_msg()
+            tm.write_stdin("whoami\r")
+            (stdout, other) = yield tm.read_stdout()
+            self.assertEqual(stdout[:6], "whoami")
+            self.assertEqual(other, [])
+            tm.close()
+
+class NamedTermTests(TermTestCase):
+    @tornado.testing.gen_test
+    def test_no_name(self):
+        with self.assertRaises(HTTPError) as context:
+            tm = yield self.get_term_client('/named/')
+            yield tm.read_msg()
+
+        # Not found
+        self.assertEqual(context.exception.code, 404)
+        self.assertEqual(context.exception.response.code, 404)
+
+    def test_new(self):
+        response = self.fetch("/new", follow_redirects=False)
+        self.assertEqual(response.code, 302)
+        url = response.headers["Location"]
+
+        # Check that the given terminal name works
+        tm = self.get_term_client(url)

```

Comment:
This will work even if a terminal wasn't created. Is it better to interrogate the NamedTermManager (which could be attached to the test case object) to check that a terminal by that name was actually created?